test/catalog/catalog_test.cpp

```diff
@@ -0,0 +1,75 @@
+#include "catalog/catalog.h"
+#include <algorithm>
+#include <random>
+#include <vector>
+#include "catalog/catalog_defs.h"
+#include "catalog/database_handle.h"
+#include "transaction/transaction_manager.h"
+#include "util/test_harness.h"
+#include "util/transaction_test_util.h"
+namespace terrier {
+
+struct CatalogTests : public TerrierTest {
+  void SetUp() override {
+    TerrierTest::SetUp();
+    txn_manager_ = new transaction::TransactionManager(&buffer_pool_, true, LOGGING_DISABLED);
+
+    txn_ = txn_manager_->BeginTransaction();
+    catalog_ = new catalog::Catalog(txn_manager_, txn_);
+  }
+
+  void TearDown() override {
+    txn_manager_->Commit(txn_, TestCallbacks::EmptyCallback, nullptr);
+
+    TerrierTest::TearDown();
+    delete catalog_;  // need to delete catalog_first
+    delete txn_manager_;
+    delete txn_;
+  }
+
+  catalog::Catalog *catalog_;
+  storage::RecordBufferSegmentPool buffer_pool_{100, 100};
+
+  transaction::TransactionContext *txn_ = nullptr;
+  transaction::TransactionManager *txn_manager_;
+};
+
+// Tests for higher level catalog API
+// NOLINTNEXTLINE
+TEST_F(CatalogTests, CreateDatabaseTest) {
```

Comment:
All the tests in this file do not have explicit checks, but only dump the catalog content to the terminal. What do we expect to see in those dumps? Can we add automated checks for them?

Reply:
I've added some checks to the test, but more can be done. This was useful, a couple of items related to table creation and deletion were missing. There is more (known) work for deletion to follow this PR.